lib/pmdtester/runner.rb

```diff
@@ -39,14 +45,28 @@ def run_local_mode
         HtmlReportBuilder.new.build(project, report_diffs)
       end
       puts 'Built all difference reports successfully!'
+      puts ''
     end
 
     def run_online_mode
       # TODO
     end
 
     def run_single_mode
-      # TODO
+      puts "Mode: #{@options.mode}"
+      puts "Patch branch/tag name: #{@options.patch_branch}"
+      puts "Patch config path: #{@options.patch_config}"
+
+      PmdReportBuilder
+        .new(@options.patch_config, @projects, @options.local_git_repo, @options.patch_branch)
+        .build
+      @projects.each do |project|
+        report_diffs = DiffBuilder.new.build_single(project.pmd_reports[@options.patch_branch])
```

Comment:
If the build_single and build methods are the same, you can extract the loop in run_local_mode and run_single_mode into a own method.